src/daemon/screenlock.vala

```diff
@@ -0,0 +1,312 @@
+                * - /usr/share/budgie-desktop/[gtklock.ini | gtklock.css]
+                */
+                string[] system_configs = {
+                    @"file://$(Budgie.CONFDIR)/budgie-desktop/gtklock.ini",
+                    @"file://$(Budgie.DATADIR)/budgie-desktop/gtklock.ini"
+                };
+
+                foreach (string? filepath in system_configs) {
+                    File file = File.new_for_uri(filepath);
+                    bool tmp = file.query_exists();
+                    if (tmp) {
+                        output += " -c " + file.get_path();
+                        break;
+                    }
+			    }
+
+                string[] style_configs = {
+                    @"file://$(Budgie.CONFDIR)/budgie-desktop/gtklock.css",
+                    @"file://$(Budgie.DATADIR)/budgie-desktop/gtklock.css"
+                };
+
+                foreach (string? filepath in style_configs) {
+                    File file = File.new_for_uri(filepath);
+                    bool tmp = file.query_exists();
+                    if (tmp) {
+                        output += " -s " + file.get_path();
+                        break;
+                    }
+			    }
+            }
+
+            return output;
+        }
+
+        private string calculate_sleep() {
+            string output = "";
+
+            // number of seconds on ac power that is inactive before action taken; 0 is never
+            int sleep_inactive_ac_timeout = this.power.get_int("sleep-inactive-ac-timeout");
+
+            // BCC sets "nothing" or "suspend" when toggled
+            string sleep_inactive_ac_type = this.power.get_string("sleep-inactive-ac-type");
+
+            // number of seconds on battery that is inactive before action taken; 0 is never
+            int sleep_inactive_battery_timeout = this.power.get_int("sleep-inactive-battery-timeout");
+
+            // BCC sets "nothing" or "suspend" when toggled
+            string sleep_inactive_battery_type = this.power.get_string("sleep-inactive-battery-type");
+
+            if (sleep_inactive_battery_type == "suspend" && sleep_inactive_battery_timeout != 0) {
+                output = "timeout " + sleep_inactive_battery_timeout.to_string();
+                output += " 'if dbus-send --print-reply=literal --dest=org.buddiesofbudgie.BudgieScreenlock /org/buddiesofbudgie/Screenlock org.buddiesofbudgie.BudgieScreenlock.OnBattery | grep \"boolean true\" > /dev/null; then systemctl suspend; fi' ";
+            }
+
+            if (sleep_inactive_ac_type == "suspend" && sleep_inactive_ac_timeout !=0) {
+                output += " timeout " + sleep_inactive_ac_timeout.to_string();
+                output += " 'if dbus-send --print-reply=literal --dest=org.buddiesofbudgie.BudgieScreenlock /org/buddiesofbudgie/Screenlock org.buddiesofbudgie.BudgieScreenlock.OnBattery | grep \"boolean false\" > /dev/null; then systemctl suspend; fi' ";
```

Comment:
Same here wrt `loginctl`.

Reply:
`loginctl suspend` returns unknown command verb 'suspend'.  The man page isnt really revealing here.  Any ideas what the suspend equivalent is?

Reply:
Ah, `loginctl suspend` is elogind-only, but there is a dbus method that we can call instead like we do in the power dialogs. See here: https://github.com/BuddiesOfBudgie/budgie-desktop/blob/35f338af03a269652ffb9e0fa2e5203f159ee7ba/src/dialogs/power/dbus.vala#L32